server/controllers/refer/courseParticipationsController.ts

```diff
@@ -48,6 +48,29 @@ export default class CourseParticipationsController {
     }
   }
 
+  details(): TypedRequestHandler<Request, Response> {
+    return async (req, res) => {
+      TypeUtils.assertHasUser(req)
+      const { courseParticipationId, referralId } = req.params
+
+      await this.courseService.getParticipation(req.user.token, courseParticipationId)
+
+      const referral = await this.referralService.getReferral(req.user.token, referralId)
+      const person = await this.personService.getPerson(req.user.username, referral.prisonNumber)
+
+      if (!person) {
```

Comment:
Wonder if we could also move this to the service at some point to avoid the repetition in each of the controllers?

Reply:
Yeah I think that'd be a sensible plan. Will add a card to the backlog for this.